examples/speech_to_text/visualize.py

```diff
@@ -0,0 +1,80 @@
+import os
+import pandas as pd
+import argparse
+import pprint
+
+
+def read_scores_from_folder(folder_path):
+    score_file_path = os.path.join(folder_path, "scores.tsv")
+    if os.path.isfile(score_file_path):
+        with open(score_file_path, "r") as f:
+            contents = [line.strip() for line in f.read().split("\n") if line.strip()]
+        return contents
+    else:
+        return None
+
+
+def read_scores_files(output_folder):
+    all_contents = []
+
+    if not os.path.isdir(output_folder):
+        raise ValueError("Output folder does not exist")
+
+    output_folder = os.path.abspath(output_folder)
+
+    for folder in os.listdir(output_folder):
+        folder_path = os.path.join(output_folder, folder)
+
+        if os.path.isdir(folder_path):
+            contents = read_scores_from_folder(folder_path)
+            if contents:
+                all_contents.append(contents)
+    return all_contents
+
+
+def process_result(output_folder, metric_names):
+    all_contents = read_scores_files(output_folder)
+
+    # Extracting headers from the first line of each "scores.tsv" file
+    headers = [contents[0].split() for contents in all_contents if contents]
+
+    if not headers:
+        raise ValueError("No headers found in the results")
```

Comment:
It might be a good idea to check for the headers inside `read_scores_files` so you can include the path to the file with no headers in the error message; this would give a more informative error message that would be helpful for debugging.